packages/navigator/src/browser/navigator-widget.ts

```diff
@@ -22,20 +25,41 @@ export const CLASS = 'theia-Files';
 @injectable()
 export class FileNavigatorWidget extends FileTreeWidget {
 
+    protected readonly disposables = new DisposableCollection();
+    protected readonly searchBox: SearchBoxWidget;
+
     constructor(
         @inject(TreeProps) readonly props: TreeProps,
         @inject(FileNavigatorModel) readonly model: FileNavigatorModel,
         @inject(ContextMenuRenderer) contextMenuRenderer: ContextMenuRenderer,
         @inject(CommandService) protected readonly commandService: CommandService,
         @inject(SelectionService) protected readonly selectionService: SelectionService,
         @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService,
-        @inject(LabelProvider) protected readonly labelProvider: LabelProvider
+        @inject(LabelProvider) protected readonly labelProvider: LabelProvider,
+        @inject(FileNavigatorSearch) protected readonly navigatorSearch: FileNavigatorSearch,
+        @inject(SearchBoxWidgetFactory) protected readonly searchBoxFactory: SearchBoxWidgetFactory
     ) {
         super(props, model, contextMenuRenderer);
         this.id = FILE_NAVIGATOR_ID;
         this.title.label = LABEL;
         this.addClass(CLASS);
         this.initialize();
+        this.searchBox = searchBoxFactory(SearchBoxWidgetProps.DEFAULT);
+        this.disposables.pushAll([
```

Comment:
just fyi, you can reuse `toDispose` from the superclass

Reply:
OK. I was looking for that.